pallets/deitos/src/lib.rs

```diff
@@ -113,54 +135,180 @@ pub mod pallet {
-            /// The account who set the new value.
-            who: T::AccountId,
+        IPRegistered {
+            ip: T::AccountId,
+            price_storage_per_block: BalanceOf<T>,
+            total_storage: Storage,
+        },
+        IPStorageUpdated {
+            ip: T::AccountId,
+            price_storage_per_block: BalanceOf<T>,
+            total_storage: Storage,
+        },
+        IPStatusChanged {
+            ip: T::AccountId,
+            status: IPStatus,
+        },
+        IPShutdown {
+            ip: T::AccountId,
         },
     }
 
     /// information.
     #[pallet::error]
     pub enum Error<T> {
         /// The value retrieved was `None` as no value was previously set.
-        NoneValue,
-        StorageOverflow,
         NonExistentStorageValue,
+        /// Math overflow
+        Overflow,
+        /// Insufficient storage
+        InsufficientStorage,
+        /// On going agreements
+        OnGoingAgreements,
     }
 
     #[pallet::call]
     impl<T: Config> Pallet<T> {
         #[pallet::call_index(0)]
         #[pallet::weight(T::WeightInfo::register_ip())]
         /* to add more parameters*/
-        pub fn register_ip(origin: OriginFor<T>, price_unit: BalanceOf<T>) -> DispatchResult {
+        pub fn register_ip(
+            origin: OriginFor<T>,
+            price_storage_per_block: BalanceOf<T>,
+            total_storage: Storage,
+        ) -> DispatchResult {
             // Check that the extrinsic was signed and get the signer.
-            let who = ensure_signed(origin)?;
+            let ip = ensure_signed(origin)?;
+            let bounding_amount = IPBoundingAmount::<T>::get()?;
+
+            T::Currency::hold(&HoldReason::Transfer.into(), &ip, bounding_amount)?;
+
+            let ip_details = InfraProviderDetails::<T> {
+                price_storage_per_block,
+                total_storage,
+                reserved_storage: Zero::zero(),
+                status: IPStatus::Validating,
+                active_agreements: BoundedVec::new(),
+            };
+
+            InfrastructureProvider::<T>::insert(&ip, ip_details);
+
+            Self::deposit_event(Event::IPRegistered {
+                ip,
+                price_storage_per_block,
+                total_storage,
+            });
 
             Ok(())
         }
 
+        /// This is a temporary call to manage the IP status.
+        /// Statuses updates should be done automatically after an environment software check.
         #[pallet::call_index(1)]
-        #[pallet::weight(T::WeightInfo::update_ip_details())]
-        pub fn update_ip_details(origin: OriginFor<T>, price_unit: BalanceOf<T>) -> DispatchResult {
-            // Check that the extrinsic was signed and get the signer.
-            let who = ensure_signed(origin)?;
+        #[pallet::weight(T::WeightInfo::register_ip())]
+        pub fn update_ip_status(
+            origin: OriginFor<T>,
+            provider: AccountIdLookupOf<T>,
```

Comment:
I think we should use a consistent naming either it is `provider`, `ip` or `infrastructure_provider`

Reply:
if you agree i can use the parameter name of the call as `infrastructure_provider` and then use `ip`  when the `ensure_origin` returns the `T::AccountId`

Reply:
How we call internal variables isn't very critical. I am more worried about the public interface. So in transaction name it is called `ip`, but the parameter where I specify an IP is called `provider`. It is not very consistent. I think that if the tx is called `update ip` the parameter that specifies this IP should be called `ip` as well.